pkg/encoding/encoding.go

```diff
@@ -81,14 +81,20 @@ func Convert(inMediaType, outMediaType string, in []byte, out io.Writer) (*runti
 		return nil, err
 	}
 
-	obj, err := runtime.Decode(inCodec, in)
-	if err != nil {
-		return nil, fmt.Errorf("error decoding from %s: %s", inMediaType, err)
-	}
+	var encoded []byte
+	if (inMediaType == outMediaType) {
+		// Assumes that the stored version is "correct". Primarily a short cut to allow CRDs to work.
+		encoded = in
+	} else {
+		obj, err := runtime.Decode(inCodec, in)
```

Comment:
Can you also move typeMeta, inCodec and outCodec creation inside this block?

Reply:
Actually It is better to just check for this case at the start of the method and do a out.Write(in) and return. It would be less nested blocks.

Reply:
Can't move typeMeta in as its used in the return. Otherwise folded in the suggested change.